Inc/ST-LIB_HIGH/Protections/Boundary.hpp

```diff
@@ -57,57 +65,88 @@ template<class Type, ProtectionType Protector> struct Boundary;
 template<class Type>
 struct Boundary<Type, BELOW> : public BoundaryInterface{
 	static constexpr ProtectionType Protector = BELOW;
+	bool has_warning_level{false};
 	Type* src = nullptr;
 	Type boundary;
-	// we have to do this because we cannot take address of rvalue (ProtectionType::BELOW)
-	
-	
-	Boundary(Type boundary):boundary(boundary)
-	{
-		
-	}
+	Type warning_threshold;
+	constexpr Boundary(const Type warn, const Type bound)
+	: has_warning_level{true}, boundary(bound), warning_threshold(warn){
+		// i havent been able to find a way to do a static_assertion.
+		if(warn < bound){
+			ErrorHandler("Warning threshold is below boundary");
+		}
+	};
+	Boundary(Type boundary):boundary(boundary){}
 	Boundary(Type* src, Boundary<Type, Protector> boundary): 
+		has_warning_level(boundary.has_warning_level),
 		src(src),boundary(boundary.boundary)
 		{
+			
+			// we have to do this because we cannot take address of rvalue (ProtectionType::BELOW)
 			boundary_type_id = Protector;
 			format_id = BoundaryInterface::format_look_up.at(type_id<Type>);
 			// we have to preallocate space, otherwise the might get moved around, invalidating the pointer, better safe than sorry
 			name.reserve(NAME_MAX_LEN);
-			message = new HeapOrder(uint16_t{111},&format_id,&boundary_type_id,&string_len,&name,&this->boundary,this->src,
+			if(this->has_warning_level){
+				warning_threshold = boundary.warning_threshold;
+				message = new HeapOrder(uint16_t{111},&format_id,&boundary_type_id,&has_warning_level,&string_len,&name,&this->warning_threshold,&this->boundary,this->src,
```

Comment:
i do not get how when sending this message a distinction between fault and warning is specified, yes there is a has_warning level bool, however it can still be a fault

Reply:
if it does not have a warning level then its a fault 100%

if it has warning level its just about checking src with warning and bound on the control station side